design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,101 @@
+  - `rook-num-bucket-storage-classes`
+- `rook-num-storage-class-device-sets` - Number of storage class device sets.
+- `rook-num-portable-storage-class-device-sets` - .
+- `rook-mons-are-pvc` - true/false whether mons are on PVC.
+- `rook-network-provider` - The network provider used for the cluster. e.g., default, host, multus.
+- `rook-external-mode` - true/false if the cluster is in external mode.
+
+## Information Rook is interested in that is already collected by Ceph telemetry
+- RBD pools (name is stripped), some config info, and the number of them.
+- Ceph Filesystems and MDS info.
+- RGW count, zonegroups, zones.
+- RBD mirroring info.
+
+## Metrics that Rook would like to see added to Ceph's telemetry
+- Number of NFS clusters
+  - Ceph can count NFS clusters by listing the namespaces in the `.nfs` pool and report those
+    namespaces that have `conf-nfs.<namespace>` objects present
+- Filesystem mirroring info
+- RGW bucket notification usage (Yuval may have good detail on this point)
+
+## Proposed integration strategy
+- Rook will first merge [`rook-version`](https://github.com/rook/rook/pull/10161).
+- Rook, with input from Ceph telemetry, will approve a version of this design doc and treat all
+  noted metric names as requirements.
+- Ceph will add all of the noted telemetry keys from the config-key database to its telemetry and
+  backport to Ceph Pacific and Ceph Quincy.
+  - Ceph code should handle the case where Rook has not set these fields.
+  - Ceph code can add `rook-version` immediately if they expect it might get into the next
+    Pacific/Quincy release more quickly.
+- Rook will implement each metric (or related metric group) individually over time.
+
+This strategy will allow Rook time to add telemetry items as it is able without rushing. Because the
+telemetry fields will be approved all at once, it will also minimize the coordination that is
+required between Ceph and Rook. The Ceph team will not need to create PRs one-to-one with Rook, and
+we can limit version mismatch issues as the telemetry is being added.
+
+## Rejected metrics
+### Count of each CR Kind
+Count of each type of CR: cluster, object, file, object store, mirror, bucket topic,
+bucket notification, etc.
+
+This was rejected for version one for a few reasons:
+1. In most cases, the Ceph telemetry should be able to report this implicitly via its information.
+  - Block pools, filesystems, and object stores can already be inferred easily
+1. It would require a config-key for each CR sub-resource, and Rook has many.
+1. This could be simpler if we had a json blob for the value, but the Ceph telemetry team has set a
+   guideline not to do this.
+
+We can revisit this on a case-by-case basis for specific CRs or features. For example, we may wish
+to have ideas about COSI usage when that is available.
+
+### Pod memory/CPU requests/limits
+The memory/CPU requests/limits set on Ceph daemon types.
+
+This was rejected for a few reasons:
```

Comment:
I don't see it as rejected or accepted yet, I'd suggest it still needs more discussion. I don't think the ceph telemetry prohibits us from collecting since it's really a small list in reality where the same resource limits are generally applied for all OSDs. Maybe we even ignore the resources set on storageClassDeviceSets and just collect the general `spec.resources`. This data doesn't have to be exhaustive, it's just to give us an idea of how features are being used and perhaps give us input to how we might change the defaults or implement features in the future.

Reply:
Laura specifically told us on the meeting yesterday not to put data structures into the config-key database. We need to clear anything like this with the telemetry team.

Reply:
I do think this would be good to consider, so perhaps it's just how to format the metrics, even if we have a number of individual metrics for memory/cpu requests/limits for mons, osds, mgr, etc.

Reply:
Can you please share an example of exactly how such metrics look like?
If we format the metrics in a way that it does not grow dynamically, and it is not too big, it should be fine.

Reply:
We discussed offline that this may not be very useful to gather, so we will hold off on it for now anyway.